### PR TITLE
Fix ios functional tests failed

### PR DESCRIPTION
**Pull request by ki4070ma**
For https://github.com/appium/python-client/issues/373

## Changes
1. Move non test files
1. Delete uiautomation test
   * since it's deprecated
1. Fix some parameters to pass tests

## Results

```
➜  ios git:(fix-ios-func-test-chrome) ✗ py.test *.py                                                                       
.........................                                                                                                                                                    [100%]

---------- coverage: platform darwin, python 3.7.3-final-0 -----------
Name                                                                                                                                                          Stmts   Miss  Cover
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/__init__.py                                       6      1    83%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/common/__init__.py                                1      0   100%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/common/exceptions.py                              3      0   100%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/common/helper.py                                 11      5    55%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/common/logger.py                                  9      0   100%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/saucetestcase.py                                 30     24    20%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/version.py                                        1      0   100%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/__init__.py                             3      0   100%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/appium_connection.py                    8      0   100%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/applicationstate.py                     6      0   100%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/clipboard_content_type.py               4      0   100%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/common/__init__.py                      1      0   100%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/common/mobileby.py                     11      0   100%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/common/multi_action.py                 23     16    30%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/common/touch_action.py                 54     40    26%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/errorhandler.py                        11      4    64%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/__init__.py                  0      0   100%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/action_helpers.py           46     37    20%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/android/__init__.py          0      0   100%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/android/activities.py       24     13    46%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/android/gsm.py              46     15    67%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/android/network.py          15      5    67%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/android/performance.py      13      5    62%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/android/power.py            13      4    69%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/android/sms.py               8      2    75%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/android/system_bars.py       7      1    86%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/applications.py             56     26    54%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/clipboard.py                21      8    62%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/context.py                  13      1    92%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/device_time.py              12      4    67%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/hw_actions.py               33      2    94%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/images_comparison.py        14      6    57%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/ime.py                      22      8    64%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/keyboard.py                 44     19    57%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/location.py                 16      6    62%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/remote_fs.py                29      8    72%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/screen_record.py            16      4    75%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/search_context.py           22      9    59%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/settings.py                 12      4    67%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/mobilecommand.py                       70      0   100%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/switch_to.py                            5      1    80%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/webdriver.py                          198     60    70%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/webelement.py                          61     14    77%
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
TOTAL                                                                                                                                                           998    352    65%

25 passed in 470.86 seconds

```

### CODE REVIEW

**Comment by ki4070ma** [comment]
Will replace app with the latest UICatalog in this PR

https://github.com/appium/ios-uicatalog

**Comment by ki4070ma** [comment]
## Results

```
➜  ios git:(fix-ios-func-test-chrome) ✗ py.test *.py
.........................                                                                                                                                                    [100%]

---------- coverage: platform darwin, python 3.7.3-final-0 -----------
Name                                                                                                                                                          Stmts   Miss  Cover
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/__init__.py                                       6      1    83%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/common/__init__.py                                1      0   100%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/common/exceptions.py                              3      0   100%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/common/helper.py                                 11      5    55%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/common/logger.py                                  9      0   100%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/saucetestcase.py                                 30     24    20%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/version.py                                        1      0   100%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/__init__.py                             3      0   100%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/appium_connection.py                    8      0   100%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/applicationstate.py                     6      0   100%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/clipboard_content_type.py               4      0   100%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/common/__init__.py                      1      0   100%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/common/mobileby.py                     11      0   100%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/common/multi_action.py                 23     16    30%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/common/touch_action.py                 54     15    72%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/errorhandler.py                        11      4    64%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/__init__.py                  0      0   100%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/action_helpers.py           46     31    33%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/android/__init__.py          0      0   100%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/android/activities.py       24     13    46%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/android/gsm.py              46     15    67%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/android/network.py          15      5    67%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/android/performance.py      13      5    62%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/android/power.py            13      4    69%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/android/sms.py               8      2    75%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/android/system_bars.py       7      1    86%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/applications.py             56     26    54%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/clipboard.py                21      8    62%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/context.py                  13      1    92%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/device_time.py              12      4    67%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/hw_actions.py               33      2    94%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/images_comparison.py        14      6    57%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/ime.py                      22      8    64%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/keyboard.py                 44     19    57%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/location.py                 16      6    62%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/remote_fs.py                29      8    72%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/screen_record.py            16      4    75%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/search_context.py           22      9    59%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/extensions/settings.py                 12      4    67%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/mobilecommand.py                       70      0   100%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/switch_to.py                            5      1    80%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/webdriver.py                          198     60    70%
/Users/atsushimori/.pyenv/versions/3.7.3/lib/python3.7/site-packages/Appium_Python_Client-0.42-py3.7.egg/appium/webdriver/webelement.py                          61     14    77%
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
TOTAL                                                                                                                                                           998    321    68%

25 passed in 500.60 seconds
```

**Comment by ki4070ma** [comment]
@KazuCocoa @mykola-mokhnach 
I've replaced UICatalog.app.zip, but it's getting bigger. (+18.2MB)
I built below repository by `npm install` and zip `build/Release-iphonesimulator/UICatalog-iphonesimulator.app`. If any problem on my steps, please let me know.

https://github.com/appium/ios-uicatalog

File: test/functional/ios/appium_tests.py [hunk 1]
```diff
@@ -100,11 +97,9 @@ def test_hide_keyboard_presskey_strategy(self):
         self.assertFalse(el.is_displayed())
 
     def test_hide_keyboard_no_key_name(self):
-        el = self.driver.find_element_by_name('Uses of UITextField')
-        el.click()
+        self._move_to_textbox()
 
-        # get focus on text field, so keyboard comes up
-        el = self.driver.find_element_by_class_name('UIATextField')
+        el = self.driver.find_elements_by_ios_predicate('wdType == "XCUIElementTypeTextField"')[0]
```
Comment (mykola-mokhnach) [inline test/functional/ios/appium_tests.py]: why not to use find_element?

Comment (ki4070ma) [reply test/functional/ios/appium_tests.py]: Explicitly to show more than one `XCUIElementTypeTextField`.
That's it, behavior is same to `find_element`.
(If you have further comments on my idea, it's welcome.)

File: test/functional/ios/appium_tests.py [hunk 2]
```diff
@@ -85,11 +84,9 @@ def test_hide_keyboard(self):
         self.assertFalse(el.is_displayed())
 
     def test_hide_keyboard_presskey_strategy(self):
-        el = self.driver.find_element_by_name('Uses of UITextField')
-        el.click()
+        self._move_to_textbox()
 
-        # get focus on text field, so keyboard comes up
-        el = self.driver.find_element_by_class_name('UIATextField')
+        el = self.driver.find_elements_by_ios_predicate('wdType == "XCUIElementTypeTextField"')[0]
```
Comment (KazuCocoa) [inline test/functional/ios/appium_tests.py]: Is it same as `find_element/s_by_class('XCUIElementTypeTextField')`?
(Other same predicates are also)

File: test/functional/ios/helper/desired_capabilities.py [hunk 3]
```diff
@@ -28,8 +28,8 @@ def get_desired_capabilities(app):
     desired_caps = {
         'deviceName': iphone_device_name(),
         'platformName': 'iOS',
-        'platformVersion': '12.1',
-        'app': PATH('../../apps/' + app),
+        'platformVersion': '12.2',
+        'app': PATH('../../../apps/' + app),
```
Comment (KazuCocoa) [inline test/functional/ios/helper/desired_capabilities.py]: Can replace `+` with `format`

Comment (ki4070ma) [reply test/functional/ios/helper/desired_capabilities.py]: Thanks, it should be.

File: test/functional/ios/appium_tests.py (hunk 2 above)
Comment (ki4070ma) [reply test/functional/ios/appium_tests.py]: Thanks, I reverted to `_by_class_name`.
(I'm not familiar with `iOS` for now, so any feedbacks are welcome.)

**Pull request merged**